File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans =""; 
        string a = ""; 
        for(auto i : s)
        {
           
            if (i==' ')
            {
                for (int l =0 ,r=a.size()-1;l<r;l++,r--)
                {
                    char t = a[l]; 
                    a[l] =a[r]; 
                    a[r] =t; 
                }
                 ans+=(a+" "); 
                a ="";
            }
            else 
            { 
                a+=i; 
               
            }
        }
         for (int l =0 ,r=a.size()-1;l<r;l++,r--)
                {
                    char t = a[l]; 
                    a[l] =a[r]; 
                    a[r] =t; 
                }
        ans+=a; 
        return ans; 
    }
};

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cs
public class Solution {
    public string ReverseWords(string s) {
        string [] arr = s.Split(' ');
        string ans=""; 
        for (int i=0;i<arr.Length;i++)
        {
            string ss= ""; 
            for (int r=arr[i].Length-1;r>=0;r--)
            {
               ss+=arr[i][r]; 
            }
            if (i==arr.Length-1)
             ans+=ss; 
             else 
             {
                 ans+=ss+" "; 
             }
        }
        
        return ans; 

    }
}